postgres-native-sqldelight-driver/src/commonMain/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriver.kt

```diff
@@ -225,6 +294,9 @@ public class PostgresNativeDriver(private var conn: CPointer<PGconn>) : SqlDrive
 
     override fun close() {
         PQfinish(conn)
+        for ((_, job) in listeners) {
+            job.cancel()
+        }
     }
```

Comment:
Should this be blocking? Now `close()` will return before the listeners are actually cancelled. The following snippet will ensure that all the listeners are finished before `close` returns. I think making it blocking also make sure that no failures disappear 🤔

```
override fun close() = runBlocking {
    PQfinish(conn)
    for ((_, job) in listeners) {
        job.cancelAndJoin()
    }
}
```

Reply:
Yeah, you are right. I really dislike `runBlocking` but I guess, we are forced to use it here to wait for cancelation.